src/dask_awkward/lib/io/io.py

```diff
@@ -460,6 +464,21 @@ def __call__(self, packed_arg):
         )
 
 
+def return_empty_on_raise(
+    fn: Callable[P, Array],
+    allowed_exceptions: tuple[type[BaseException], ...],
+) -> Callable[P, Array]:
+    @functools.wraps(fn)
+    def wrapped(*args, **kwargs):
+        try:
+            result = fn(*args, **kwargs)
+            return result
+        except allowed_exceptions:
+            return ak.Array(fn.form.length_zero_array(highlevel=False))
```

Comment:
We also have the existing `.mock` interface: https://github.com/douglasdavis/dask-awkward/blob/a368106dd57c4a355022db42ab9f97f853e2f566/src/dask_awkward/layers/layers.py#L54-L56

I wonder if either we should extend this with another method `mock_empty()`, or whether we should require only `mock` and take the form from the mocked array: `fn.mock().layout.form.length_zero_array()`

One thing to be mindful of is that we should allow the implementer to choose the backend.

Reply:
Ah yes I like the idea of reusing mock for this task, thanks for peaking at this

Reply:
Since `mock` has been around for a while and has been a typetracer-only part of the code, perhaps it is best to go with a new `mock_empty(backend: str)` method

Reply:
Concerning the backend configurability: we should probably forbid the implementer to inject the typetracer backend as for this purpose it just doesn't make any sense?